### PR TITLE
Add ISTFT

### LINKED ISSUE #358: ISTFT [FEA]
ISTFT is mentioned in the documentation under STFT, but hasn't been implemented. Are there plans to add ISTFT? 

Thanks,
TIm

### PR DESCRIPTION
**Pull request by mnicely**
Closes #358 

Initial attempt at adding ISTFT.
Needs further improvements.

A100 PCIe 40GB
```bash
------------------- benchmark 'ISTFT': 16 tests --------------------
Name (time in ms)                                     Mean          
--------------------------------------------------------------------
test_istft_cpu[2048-1000000.0-65536-float64]        1.7042 (1.0)    
test_istft_cpu[2048-1.0-65536-float64]              1.7094 (1.00)   
test_istft_cpu[1024-1000000.0-65536-float64]        1.8359 (1.08)   
test_istft_cpu[1024-1.0-65536-float64]              1.8685 (1.10)   
test_istft_gpu[2048-1.0-65536-float64]              3.4186 (2.01)   
test_istft_gpu[2048-1000000.0-65536-float64]        3.4438 (2.02)   
test_istft_gpu[2048-1.0-65536-complex128]           3.4542 (2.03)   
test_istft_cpu[1024-1000000.0-65536-complex128]     3.5705 (2.10)   
test_istft_cpu[1024-1.0-65536-complex128]           3.6211 (2.12)   
test_istft_gpu[2048-1000000.0-65536-complex128]     4.1159 (2.42)   
test_istft_cpu[2048-1000000.0-65536-complex128]     4.9498 (2.90)   
test_istft_cpu[2048-1.0-65536-complex128]           4.9533 (2.91)   
test_istft_gpu[1024-1000000.0-65536-float64]        6.3531 (3.73)   
test_istft_gpu[1024-1.0-65536-float64]              6.3744 (3.74)   
test_istft_gpu[1024-1000000.0-65536-complex128]     6.4064 (3.76)   
test_istft_gpu[1024-1.0-65536-complex128]           6.4077 (3.76)   
--------------------------------------------------------------------
```

### CODE REVIEW

File: python/cusignal/spectral_analysis/spectral.py [hunk 1]
```diff
@@ -1040,6 +1040,261 @@ def stft(
...
+    >>> plt.show()
+    Note that the cleaned signal does not start as abruptly as the original,
+    since some of the coefficients of the transient were also removed:
+    >>> plt.figure()
+    >>> plt.plot(time, x, time, xrec, time, carrier)
+    >>> plt.xlim([0, 0.1])
+    >>> plt.xlabel('Time [sec]')
+    >>> plt.ylabel('Signal')
+    >>> plt.legend(['Carrier + Noise', 'Filtered via STFT', 'True Carrier'])
+    >>> plt.show()
```
Comment (awthomp) [inline python/cusignal/spectral_analysis/spectral.py:1183]: Let's update the example to cusignal. Before plotting, we'll need to run a `cp.asnumpy()`.

**Review by awthomp — CHANGES_REQUESTED** [review]

**Comment by mnicely** [comment]
Updated performance

```bash
--------------------- benchmark 'ISTFT': 16 tests ----------------------
Name (time in us)                                         Mean          
------------------------------------------------------------------------
test_istft_gpu[2048-1000000.0-65536-float64]          436.6124 (1.0)    
test_istft_gpu[2048-1.0-65536-float64]                437.3932 (1.00)   
test_istft_gpu[1024-1000000.0-65536-float64]          442.5973 (1.01)   
test_istft_gpu[1024-1.0-65536-float64]                453.8028 (1.04)   
test_istft_gpu[1024-1000000.0-65536-complex128]       454.2742 (1.04)   
test_istft_gpu[2048-1000000.0-65536-complex128]       454.8317 (1.04)   
test_istft_gpu[1024-1.0-65536-complex128]             456.1441 (1.04)   
test_istft_gpu[2048-1.0-65536-complex128]             459.9929 (1.05)   
test_istft_cpu[2048-1.0-65536-float64]              1,704.0559 (3.90)   
test_istft_cpu[2048-1000000.0-65536-float64]        1,706.9067 (3.91)   
test_istft_cpu[1024-1000000.0-65536-float64]        1,838.5234 (4.21)   
test_istft_cpu[1024-1.0-65536-float64]              1,952.8545 (4.47)   
test_istft_cpu[1024-1000000.0-65536-complex128]     3,511.4455 (8.04)   
test_istft_cpu[1024-1.0-65536-complex128]           3,650.8632 (8.36)   
test_istft_cpu[2048-1.0-65536-complex128]           4,855.2903 (11.12)  
test_istft_cpu[2048-1000000.0-65536-complex128]     4,856.6543 (11.12)  
------------------------------------------------------------------------
```

File: python/cusignal/spectral_analysis/spectral.py (hunk 1 above)
Comment (mnicely) [reply python/cusignal/spectral_analysis/spectral.py:1212]: Done

**Review by awthomp — APPROVED** [review]

**Comment by awthomp** [comment]
@gpucibot merge

**Pull request merged**